src/anm/analytics.js

```diff
@@ -0,0 +1,77 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = supportSendBeacon ? 2000 : 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics');
+
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+                setTimeout(event, timeout);
+            } else {
+                var trackUrl = animatronUrl + '?player=' + encodeURIComponent(data);
+                sendViaGif(trackUrl);
+            }
+        } else {
+            setTimeout(event, timeout);
+        }
+    };
+
+    var sendViaGif = function (trackUrl) {
+        if (!beacon) {
+            beacon = engine.createStatImg();
+        }
+        beacon.src = trackUrl;
+        beacon.onerror = beacon.onload = function () {
+            beacon.onerror = beacon.onload = null;
+            setTimeout(event, timeout);
+        }
+    };
+    event();
```

Comment:
>So, the question is "Why?" :).

Because it doesn't matter. It's obvious and I thought about it. There would be a reason to complicate the code even more if check for queue.empty() would be super expensive. You know, "premature optimization is the root of all evil".